text/0039-reduce-module-size.md

```diff
@@ -0,0 +1,550 @@
+(and increasing) CDK modules. The table below breaks down the percentage that
+each category of files contributes to the size of aws-cdk-lib.
+
+Note: The percentages do not add up to 100. Some files, e.g. various .json
+files, are excluded and contribute very little to the size.
+
+| Category                       | Percentage of aws-cdk-lib size |
+| ------------------------------ | ------------------------------ |
+| .jsii files                    | 41.64%                         |
+| Source Maps (.js.map)          | 21.93%                         |
+| Lambda Layer zip files         | 15.48%                         |
+| Javascript code (.js)          | 9.17%                          |
+| Type Declaration files (.d.ts) | 9.15%                          |
+| README, etc (.md)              | 0.76%                          |
+| bundled npm dependencies       | 0.69%                          |
+| .ts-fixture                    | 0.37%                          |
+| .jsiirc                        | 0.36%                          |
+
+### Why is the package going to be larger in the future?
+
+The CDK team is currently blocked on adding more Lambda Layer zip files to the
+AWS CDK framework, because adding them would dramatically increase the size of
+the package. Each of these zips are in a module whose only purpose is to bundle
+a dependency (or two) into a Lambda Layer, which is then used in custom
+resources that are part of the AWS CDK framework.
+
+There are more dependencies like these that need to be added to the framework,
+but we are currently blocked on adding them with the current design because they
+would increase the size too much. The current known list is:
+
+1. Additional versions of
+   [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl). This is a
+   pressing problem for customers using the EKS module, who are limited to a
+   single version of the kubectl CLI today.
+2. Additional versions of [awscli](https://pypi.org/project/awscli/). See
+   [issue](https://github.com/aws/aws-cdk/issues/13993).
+3. [Boto3](https://pypi.org/project/boto3/), and
+   [aws-sdk](https://www.npmjs.com/package/aws-sdk). All custom resources in the
+   AWS CDK framework currently rely on whatever version of the AWS SDK that is
+   available in the Lambda runtime by default. To make the custom resources more
+   robust, and allow us to choose the exact version of the AWS SDK used by them,
+   we should bundle these dependencies into each custom resource’s Lambda code.
+
+If we add all of these with the current design, that would increase the size of
+`aws-cdk-lib` by approximately 130 MiB*, and make the Lambda Layers the biggest
+contributor of size at 39%.
+
+*See Appendix A to see where this estimate came from.
+
+| Category                       | Percentage of aws-cdk-lib size |
+| ------------------------------ | ------------------------------ |
+| Lambda Layer zip files         | 38.83%                         |
```

Comment:
If this table is intended to show the new distribution AFTER hypothetical future changes, I'm not sure that's necessary. The text is sufficient, and this table confused me initially because the numbers were so much different from the previous one. 

If you do want to show this data, combine it into 1 table I suppose.

Reply:
I think this table is necessary to show why it is important to address the lambda layer zips. If not for the future things we want to add, then we could just address the .jsii files and remove the source maps and call it a day.

Reply:
I am not sure how to logically combine this into one table and still have it makes sense in the flow of the doc. I change the title of the 2nd column of this one. Does that help make it less confusing? How would you like to see it represented?

Reply:
Resolving this comment, please re-open if you have a suggestion here.